src/validation.cpp

```diff
@@ -3747,37 +3747,35 @@ bool CChainState::AcceptBlock(const std::shared_ptr<const CBlock>& pblock, CVali
     return true;
 }
 
-bool ProcessNewBlock(const CChainParams& chainparams, const std::shared_ptr<const CBlock> pblock, bool fForceProcessing, bool *fNewBlock)
+bool ProcessNewBlock(const CChainParams& chainparams, const std::shared_ptr<const CBlock> pblock, CValidationState& dos_state, bool fForceProcessing, bool *fNewBlock)
 {
     AssertLockNotHeld(cs_main);
 
     {
         CBlockIndex *pindex = nullptr;
         if (fNewBlock) *fNewBlock = false;
-        CValidationState state;
 
         // CheckBlock() does not support multi-threaded block validation because CBlock::fChecked can cause data race.
         // Therefore, the following critical section must include the CheckBlock() call as well.
         LOCK(cs_main);
 
         // Ensure that CheckBlock() passes before calling AcceptBlock, as
         // belt-and-suspenders.
-        bool ret = CheckBlock(*pblock, state, chainparams.GetConsensus());
+        bool ret = CheckBlock(*pblock, dos_state, chainparams.GetConsensus());
         if (ret) {
             // Store to disk
-            ret = ::ChainstateActive().AcceptBlock(pblock, state, chainparams, &pindex, fForceProcessing, nullptr, fNewBlock);
+            ret = ::ChainstateActive().AcceptBlock(pblock, dos_state, chainparams, &pindex, fForceProcessing, nullptr, fNewBlock);
         }
         if (!ret) {
-            GetMainSignals().BlockChecked(*pblock, state);
-            return error("%s: AcceptBlock FAILED (%s)", __func__, FormatStateMessage(state));
+            return error("%s: AcceptBlock FAILED (%s)", __func__, FormatStateMessage(dos_state));
```

Comment:
Unrelated to this commit, the error here may be from either `CheckBlock` or `AcceptBlock`. Related to this commit, the returned `CValidationState` could also be from either rather than just `AcceptBlock` but the commit summary mentions only the latter.

I think this reflects a need for better naming / code organization rather than the need to make the commit message more verbose, however.

Reply:
but since this commit is touching this code, might be worth updating error message to "AcceptBlock or CheckBlock FAILED..." to be more accurate?

Reply:
Historically, CheckBlock was only called as a part of AcceptBlock, and I think we'd like to eventually go back there. See #/9765 and the bitcoin-dev mail titled "Vulnerability relating to 64-byte transactions in Bitcoin Core 0.13 branch". (sorry, lots and lots of context to this code.....)